arwen/elrondapi/elrondei.go

```diff
@@ -1513,17 +1511,14 @@ func upgradeContract(
 	// Set up the async call as if it is not known whether the called SC
 	// is in the same shard with the caller or not. This will be later resolved
 	// by runtime.ExecuteAsyncCall().
-	callData := txDataBuilder.NewBuilder()
-	callData.Func(arwen.UpgradeFunctionName)
-	callData.Bytes(code).Bytes(codeMetadata)
-
+	callData := arwen.UpgradeFunctionName + "@" + hex.EncodeToString(code) + "@" + hex.EncodeToString(codeMetadata)
```

Comment:
There is a txDataBuilder that is currently used for tests, maybe it is worth extracting and using in production code too. Low prio

Reply:
I think this txDataBuilder will be moved to the upcoming elrond-go-core repo